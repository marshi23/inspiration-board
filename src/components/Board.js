import React, { Component } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';

import './Board.css';
import Card from './Card';
import NewCardForm from './NewCardForm';
import CARD_DATA from '../data/card-data.json';


class Board extends Component {
  constructor() {
    super();

    this.state = {
      cards: []


      // cards: CARD_DATA.cards.map((card, i) => {
      //   return <Card
      //     key={i}
      //     text={card.text}
      //     emoji={card.emoji || card.Emoji}
      //   />
      // }),
    }
  };

componentDidMount() {
  let url = this.props.url + this.props.name + '/cards'
  console.log(url)
axios.get(this.props.url+this.props.name+'/cards')

  .then((response) => {
    const cards = response.data.map((card) => {

      const newCard = {
        key: card.card.id,
        text: card.card.text,
        emoji: card.card.emoji
        // ...card.card
      };
      return newCard;
    });

    console.log(cards)
    this.setState({
      cards: cards
    });
  })
  .catch(() => {
    console.log('where to catch errors')
  })
}

displayCards = () => {
  return this.state.cards.map((card, i) => {
    return <Card
      key={i}
      id={card.id}
      text={card.text}
      emoji={card.emoji}
      />
  });
}

  render() {

    return (
      <div className="board">
        {this.displayCards()}
      </div>
    )
  }

}

Board.propTypes = {

};

export default Board;
