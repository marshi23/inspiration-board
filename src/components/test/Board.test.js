import React from 'react';
import Board from '../Board';
import { shallow } from 'enzyme';


describe('Board', () => {
  it('test it matches and exisiting snapshot', () => {
    const wrapper = shallow(
      <Board
        url="https://inspiration-board.herokuapp.com/boards/"
        boardName={"maryam"}
      />);

    expect(wrapper).toMatchSnapshot();
  });

});
