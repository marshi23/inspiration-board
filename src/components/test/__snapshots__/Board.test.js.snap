// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board test it matches and exisiting snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Board
    boardName="maryam"
    url="https://inspiration-board.herokuapp.com/boards/"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="errorMessages"
        >
          
        </section>,
        <section>
          <NewCardForm
            addCardCallback={[Function]}
          />
        </section>,
        Array [],
      ],
      "className": "board",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "errorMessages",
        },
        "ref": null,
        "rendered": "",
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NewCardForm
            addCardCallback={[Function]}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "addCardCallback": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "section",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="errorMessages"
          >
            
          </section>,
          <section>
            <NewCardForm
              addCardCallback={[Function]}
            />
          </section>,
          Array [],
        ],
        "className": "board",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "errorMessages",
          },
          "ref": null,
          "rendered": "",
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NewCardForm
              addCardCallback={[Function]}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "addCardCallback": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "section",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
